locksmith/src/utils/lockMigrate.ts

```diff
@@ -0,0 +1,260 @@
+import { ethers, Wallet } from 'ethers'
+import * as contracts from '@unlock-protocol/contracts'
+import { NetworkConfig } from '@unlock-protocol/types'
+import { networks } from '@unlock-protocol/networks'
+import EventEmitter from 'events'
+import listManagers from './lockManagers'
+
+interface LockClonerProps {
+  lockAddress: string
+  unlockVersion: Number
+  chainId: number
+  recordId: number
+}
+
+class MigrateLogEventEmitter extends EventEmitter {}
+
+export const migrateLogEvent = new MigrateLogEventEmitter()
+
+// TODO: how to pass this safely?
+const mnemonic =
+  process.env.WALLET_MNEMONIC ||
+  'test test test test test test test test test test test junk'
```

Comment:
this remains a challenge as we need to deploy from the server...

Reply:
Let's use the same credentials as the key purchaser. we already have it setup.